contracts/src/Gateway.sol

```diff
@@ -617,4 +621,21 @@ contract Gateway is IGateway, IInitializable {
         assets.assetHubCreateAssetFee = config.assetHubCreateAssetFee;
         assets.assetHubReserveTransferFee = config.assetHubReserveTransferFee;
     }
+
+    /// @inheritdoc IGateway
+    function transact(ParaID destinationChain, TransactMessage calldata message) external payable {
+        if (message.validate() == false) {
+            revert InvalidTransact();
+        }
+        Ticket memory ticket;
+        Costs memory costs;
+        costs.foreign = message.fee;
```

Comment:
To confirm what we discussed on the call:

* Similarly to Moonbeam's transact implementation, the user needs to have a prefunded sovereign account on the destination chain. The fee will be withdrawn from this account, and used to buy weight.
* Therefore, on the ethereum side, the user only needs to pay a small fee to cover the relayer reward on BridgeHub. This fee should be deposited into the agent of the destination parachain.

Reply:
https://github.com/Snowfork/polkadot-sdk/pull/114/commits/a8b84355336e9968c1ebe87054b8b465ffab3606